### PR TITLE
Standardize repo (minus tests)

### PR DESCRIPTION
**Pull request by mcmire**
* Add missing workflows from template repo
* Replace `.gitattributes`, `.editorconfig`, `.gitignore` with version
  from template repo
* Add NPM scripts in template repo but missing here
  * Add `allow-scripts` and run it via `yarn setup`
* Update TypeScript config to match template repo
  * Exclude test files
  * Use ES2020 `.d.ts` instead of ES2017
* Upgrade Node to v14
* Lint JSON files with `eslint-plugin-packagejson` instead of
  `prettier-plugin-json`
* Update README to match template repo
* Update and backfill changelog

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        node-version: [12.x]
+        node-version: [14.x, 16.x]
```
Comment (Gudahtt) [inline .github/workflows/build-test.yml:14]: This package still supports Node.js v12 (according to `package.json`), so we should continue to test it until we drop support

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Mostly looks good! Only issue I saw was the CI `build-test` Node.js version

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (mcmire) [reply .github/workflows/build-test.yml]: Good catch! Fixed in [29f4994](https://github.com/MetaMask/eth-block-tracker/pull/97/commits/29f499475854291e673ce72d861d715c38c32143).

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**